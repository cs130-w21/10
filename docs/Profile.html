<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Profile</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Profile</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Profile<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description"><p>Component to render the Profile page of any logged in user</p></div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="Profile"><span class="type-signature"></span>new Profile<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Profile_Profile.js.html">components/Profile/Profile.js</a>, <a href="components_Profile_Profile.js.html#line63">line 63</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    
        <h3 class="subsection-title">Extends</h3>

        


    <ul>
        <li>React.Component</li>
    </ul>


    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="showEditPopup"><span class="type-signature"></span>showEditPopup<span class="type-signature"> :boolean</span></h4>




<div class="description">
    <p>The state that determines whether the <a href="EditProfilePopup.html">EditProfilePopup</a> is shown or not</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type">boolean</span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Profile_Profile.js.html">components/Profile/Profile.js</a>, <a href="components_Profile_Profile.js.html#line73">line 73</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="userData"><span class="type-signature"></span>userData<span class="type-signature"> :<a href="global.html#UserData">UserData</a></span></h4>




<div class="description">
    <p>JSON object that contains all the user's info, retrieved from Firebase</p>
</div>



    <h5>Type:</h5>
    <ul>
        <li>
            
<span class="param-type"><a href="global.html#UserData">UserData</a></span>


        </li>
    </ul>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Profile_Profile.js.html">components/Profile/Profile.js</a>, <a href="components_Profile_Profile.js.html#line65">line 65</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="openEditPopup"><span class="type-signature"></span>openEditPopup<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Called to open the <a href="EditProfilePopup.html">EditProfilePopup</a> and allow user to edit their details.</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Profile_Profile.js.html">components/Profile/Profile.js</a>, <a href="components_Profile_Profile.js.html#line88">line 88</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id="openEditPopup"><span class="type-signature"></span>openEditPopup<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Called to close the <a href="EditProfilePopup.html">EditProfilePopup</a>.</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Profile_Profile.js.html">components/Profile/Profile.js</a>, <a href="components_Profile_Profile.js.html#line95">line 95</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
            

    

    
    <h4 class="name" id="setShowEditPopup"><span class="type-signature"></span>setShowEditPopup<span class="signature">(val)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>A function callback that sets the showEditPopup boolean in the internal state of Profile.</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>val</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Profile_Profile.js.html">components/Profile/Profile.js</a>, <a href="components_Profile_Profile.js.html#line80">line 80</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Modules</h3><ul><li><a href="module-useEditProfileForm.html">useEditProfileForm</a></li></ul><h3>Classes</h3><ul><li><a href="AuthProvider.html">AuthProvider</a></li><li><a href="CompleteYourProfile.html">CompleteYourProfile</a></li><li><a href="EditProfileForm.html">EditProfileForm</a></li><li><a href="EditProfilePopup.html">EditProfilePopup</a></li><li><a href="Login.html">Login</a></li><li><a href="MatchCard.html">MatchCard</a></li><li><a href="MatchPage.html">MatchPage</a></li><li><a href="Navbar.html">Navbar</a></li><li><a href="Profile.html">Profile</a></li><li><a href="ProfileCards.html">ProfileCards</a></li><li><a href="ProtectedRoute.html">ProtectedRoute</a></li><li><a href="SingleMatchPopup.html">SingleMatchPopup</a></li></ul><h3><a href="global.html">Global</a></h3>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.6</a> on Mon Mar 08 2021 00:40:32 GMT+0000 (Coordinated Universal Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>