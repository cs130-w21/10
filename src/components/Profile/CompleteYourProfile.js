import React from 'react';
import Paper from '@material-ui/core/Paper';

import EditProfileForm from './EditProfileForm';

const CompleteYourProfile = () => {
  return (
    <Paper>
      <EditProfileForm onCancel={() => {}}/>
    </Paper>
  );
};

export default CompleteYourProfile;
