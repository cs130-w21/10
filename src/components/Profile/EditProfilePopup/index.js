import TestEditProfilePopup from './TestEditProfilePopup';

export default TestEditProfilePopup;
