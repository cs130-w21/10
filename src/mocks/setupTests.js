const useAuthMock =  () => {
  let userData = {
    contactInfo: {
      email: 'test@test.com',
      linkedin: 'https://test.com'
    },
    expertises: ["Business"],
    interests: ["Business"],
    personalInfo: {
      bio: "testBio",
      education: {
        gradYear: 2021,
        major: 'CS',
        school: 'UCLA',
      },
      name: 'test user',
      profilePicture: '/placeholder.jpg',
      work: {
        company: 'testCompany',
        description: 'testDescription',
        position: 'testPosition',
      },
    },
  };
  const loading = false;
  let uid = '123456789';

  const logout = jest.fn(() => {
    uid = null;
  });
  const updateUserData = (field, newInfo) => {
    userData[field] = newInfo;
  };
  return { loading, userData, uid, logout, updateUserData };
};

export { useAuthMock };