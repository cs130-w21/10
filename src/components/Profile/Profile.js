import React, { useState } from 'react';
import Grid from '@material-ui/core/Grid';
import Divider from '@material-ui/core/Divider';
import Image from 'material-ui-image';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import EditIcon from '@material-ui/icons/Edit';
import LinkedInIcon from '@material-ui/icons/LinkedIn';
import EmailIcon from '@material-ui/icons/Email';
import { useAuth } from '../../services/AuthContext';
import EditProfilePopup from './EditProfilePopup';

const useStyles = makeStyles((theme) => ({
  grid: {
    padding: theme.spacing(5),
    paddingRight: theme.spacing(5),
    minWidth: theme.spacing(150),

  },
  profilePic: {
    width: theme.spacing(25),
    height: theme.spacing(25),
    marginBottom: theme.spacing(5),
    marginLeft: 'auto',
    marginRight: 'auto',
  },
  interestedArea: {
    textAlign: "left",
  },
  divider: {
    backgroundColor: "black",
  },
  areaTitle: {
    fontSize: "1rem",
    fontWeight: "500",
    lineHeight: "1.75",
    color: "#808080",
  },
  mainTitle: {
    fontSize: "1.4rem",
    fontWeight: "700",
    lineHeight: "1.75",
    color: "#515fb0",
  },
  areaEntry: {
    fontWeight: "400",
    fontSize: "0.9rem",
    lineHeight: "1.5",
  },
  contact: {
    display: 'flex',
    flexWrap: 'wrap',
    alignItems: 'center',
  }
}));

/**
 * @global
 * @typedef {Object} Profile
 * @prop {bool} showEditPopup        - The state that determines whether the EditProfilePopup is shown or not
 * @prop {callback} setShowEditPopup - A function callback that sets the showEditPopup boolean in the internal state of Profile
 * @prop {Object} userData           - A JSON object that contains all the user's info, retrieved from firebase
 */

/**
 * @class
 * @classdesc Component to render the Profile page of any logged in user
 * @extends React.Component
 */

const Profile = () => {
  const classes = useStyles();
  const { userData } = useAuth();
  const [showEditPopup, setShowEditPopup] = useState(false);
  /**
   * @memberof Profile
   * @function openEditPopup
   * @description Called to open the edit profile popup and allow user to edit it
   * @instance
   */
  const openEditPopup = () => setShowEditPopup(true);
    /**
   * @memberof Profile
   * @function openEditPopup
   * @description Called to close the edit profile popup
   * @instance
   */
  const closeEditPopup = () => setShowEditPopup(false);

  if (!userData) {
    return <div> Loading </div>
  }
  // just make sure these aren't null
  if (!userData.interests) {
    userData.interests = [];
  }
  if (!userData.expertises) {
    userData.expertises = [];
  }
  return (
    <>
      <EditProfilePopup isOpen={showEditPopup} onDismiss={closeEditPopup} />
      <Grid
        container
        spacing={0}
        direction="row"
        alignItems="center"
        justify="center"
      >

      
        <Grid item xs={12} sm={8} lg={8} container className={classes.grid} justify="flex-start" spacing={5}>
          <Grid item container xs={12} sm={5} lg={4} alignItems="center" justify="flex-start">
            <Grid item xs={12}>
              <Image src={userData.personalInfo.profilePicture || 'placeholder.jpg'} className={ classes.profilePic }/>
            </Grid>
            <Grid item xs={12} sm={9} md={6} lg={5} xl={4}>
              <Typography align="left" variant="subtitle2" component="h4" className={classes.areaTitle}>
                INTERESTS
              </Typography>
            </Grid>
            <Grid item xs={6} sm={3} md={6} lg={7} xl={8}>
              <Divider className={classes.divider} />
            </Grid>
            <Grid item xs={12}>
            {
            userData.interests.map((interest) => (
              <Typography key={interest} align="right" variant="body1" component="h6" className={classes.areaEntry}>
                {interest}
              </Typography>
            ))}
            <br />
            <br />
            </Grid>
            <Grid item xs={12} sm={9} md={6} lg={5} xl={4}>
            <Typography align="left" variant="subtitle2" component="h4" className={classes.areaTitle}>
              EXPERTISE
            </Typography>
              
            </Grid>
            <Grid item xs={6} sm={3} md={6} lg={7} xl={8}>
              <Divider className={classes.divider} />
            </Grid>
            <Grid item xs={12}>
            {
            userData.expertises.map((interest) => (
              <Typography key={interest} align="right" variant="body1" component="h6" className={classes.areaEntry}>
                {interest}
              </Typography>
            ))}
            </Grid>
              
          </Grid>
          <Grid item xs={12} sm={7} lg={8}>
            <Typography align="left" variant="h4" component="h4">
              {userData.personalInfo.name}
              <IconButton onClick={openEditPopup}>
                <EditIcon />
              </IconButton>
            </Typography>
            
            <Typography align="left" variant="subtitle2" component="h4" className={classes.areaTitle}>
              {userData.personalInfo.work.position}
            </Typography>
              
            <Typography align="left" variant="subtitle2" component="h4" className={classes.mainTitle}>
              BIO
            </Typography>
            <Typography paragraph align="left">
              {userData.personalInfo.bio || "Edit Profile to add your Bio!"}
            </Typography>
            <Typography align="left" variant="subtitle2" component="h4" className={classes.mainTitle}>
              WORK EXPERIENCE
            </Typography>
            <Typography align="left" variant="subtitle2" component="h4" className={classes.areaTitle}>
            {userData.personalInfo.work.company}, {userData.personalInfo.work.position}
              {/* replace this later w/ info from database */}
            </Typography>
            <Typography paragraph align="left">
            {userData.personalInfo.work.description}
            </Typography>
            <Typography align="left" variant="subtitle2" component="h4" className={classes.mainTitle}>
              EDUCATION
            </Typography>
            <Typography align="left" variant="subtitle2" component="h4" className={classes.areaTitle}>
              {userData.personalInfo.education.school}
            </Typography>
            <Typography align="left">
              {userData.personalInfo.education.major} major
            </Typography>
            <Typography align="left">
              Class of {userData.personalInfo.education.gradYear}
            </Typography>
            <Typography align="left" variant="subtitle2" component="h4" className={classes.mainTitle}>
              CONTACT INFO
            </Typography>
            <Typography align="left" className={classes.contact}>
            <EmailIcon />
              : {userData.contactInfo.email} 
            </Typography>

            
            <Typography paragraph align="left" className={classes.contact}>
              <LinkedInIcon />
              : <a href={userData.contactInfo.linkedin}> {userData.contactInfo.linkedin}</a>
            </Typography>
            
          </Grid>
        </Grid>
      </Grid>
    </>
  );
  
}

export default Profile;
