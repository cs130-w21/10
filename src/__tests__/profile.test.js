import { render, screen } from '@testing-library/react';
import Profile from '../components/Profile';
import { useAuthMock } from '../mocks/setupTests';
import renderer from 'react-test-renderer';
const auth = require('../services/AuthContext');
describe('test profile page', () => {
  let mockAuth;
  beforeEach(() => {
    mockAuth = jest.spyOn(auth,
      'useAuth').mockImplementation(useAuthMock);
  });
  afterEach(() => {
    mockAuth.mockRestore();
  });
  test('render profile: check if fields are present', () => {
    render(<Profile />);
    const fields = screen.getAllByText(/Bio/i);
    expect(fields[0]).toBeInTheDocument();
    const interest = screen.getByText(/Interest/i);
    expect(interest).toBeInTheDocument();
    const interestField = screen.getByText(/Business/i);
    const expertiseField = screen.getByText(/Agriculture/i);
    expect(interestField).toBeInTheDocument();
    expect(expertiseField).toBeInTheDocument();
  });
  test('snapshot test for profile page', () => {
    const tree = renderer.create(<Profile />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});