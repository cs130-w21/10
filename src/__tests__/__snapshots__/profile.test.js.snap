// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test profile page snapshot test for profile page 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root makeStyles-grid-9 MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-lg-8"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-lg-4"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          style={
            Object {
              "backgroundColor": "#fff",
              "paddingTop": "calc(1 / 1 * 100%)",
              "position": "relative",
            }
          }
        >
          <img
            className="makeStyles-profilePic-10"
            onError={[Function]}
            onLoad={[Function]}
            src="/placeholder.jpg"
            style={
              Object {
                "filterBrightness": 0,
                "filterSaturate": 20,
                "height": "100%",
                "left": 0,
                "objectFit": "inherit",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "
        filterBrightness 2250ms cubic-bezier(0.4, 0.0, 0.2, 1),
        filterSaturate 3000ms cubic-bezier(0.4, 0.0, 0.2, 1),
        opacity 1500ms cubic-bezier(0.4, 0.0, 0.2, 1)",
                "width": "100%",
              }
            }
          />
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "height": "100%",
                "justifyContent": "center",
                "left": 0,
                "pointerEvents": "none",
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          >
            <div
              className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style={
                Object {
                  "height": 48,
                  "width": 48,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9 MuiGrid-grid-md-6 MuiGrid-grid-lg-5 MuiGrid-grid-xl-4"
      >
        <h4
          className="MuiTypography-root makeStyles-areaTitle-13 MuiTypography-subtitle2 MuiTypography-alignLeft"
        >
          INTERESTS
        </h4>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3 MuiGrid-grid-md-6 MuiGrid-grid-lg-7 MuiGrid-grid-xl-8"
      >
        <hr
          className="MuiDivider-root makeStyles-divider-12"
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h6
          className="MuiTypography-root makeStyles-areaEntry-15 MuiTypography-body1 MuiTypography-alignRight"
        >
          Business
        </h6>
        <br />
        <br />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9 MuiGrid-grid-md-6 MuiGrid-grid-lg-5 MuiGrid-grid-xl-4"
      >
        <h4
          className="MuiTypography-root makeStyles-areaTitle-13 MuiTypography-subtitle2 MuiTypography-alignLeft"
        >
          EXPERTISE
        </h4>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3 MuiGrid-grid-md-6 MuiGrid-grid-lg-7 MuiGrid-grid-xl-8"
      >
        <hr
          className="MuiDivider-root makeStyles-divider-12"
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h6
          className="MuiTypography-root makeStyles-areaEntry-15 MuiTypography-body1 MuiTypography-alignRight"
        >
          Agriculture
        </h6>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-7 MuiGrid-grid-lg-8"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft"
      >
        test user
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </span>
        </button>
      </h4>
      <h4
        className="MuiTypography-root makeStyles-areaTitle-13 MuiTypography-subtitle2 MuiTypography-alignLeft"
      >
        testPosition
      </h4>
      <h4
        className="MuiTypography-root makeStyles-mainTitle-14 MuiTypography-subtitle2 MuiTypography-alignLeft"
      >
        BIO
      </h4>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph MuiTypography-alignLeft"
      >
        testBio
      </p>
      <h4
        className="MuiTypography-root makeStyles-mainTitle-14 MuiTypography-subtitle2 MuiTypography-alignLeft"
      >
        WORK EXPERIENCE
      </h4>
      <h4
        className="MuiTypography-root makeStyles-areaTitle-13 MuiTypography-subtitle2 MuiTypography-alignLeft"
      >
        testCompany
        , 
        testPosition
      </h4>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph MuiTypography-alignLeft"
      >
        testDescription
      </p>
      <h4
        className="MuiTypography-root makeStyles-mainTitle-14 MuiTypography-subtitle2 MuiTypography-alignLeft"
      >
        EDUCATION
      </h4>
      <h4
        className="MuiTypography-root makeStyles-areaTitle-13 MuiTypography-subtitle2 MuiTypography-alignLeft"
      >
        UCLA
      </h4>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft"
      >
        CS
         major
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft"
      >
        Class of 
        2021
      </p>
      <h4
        className="MuiTypography-root makeStyles-mainTitle-14 MuiTypography-subtitle2 MuiTypography-alignLeft"
      >
        CONTACT INFO
      </h4>
      <p
        className="MuiTypography-root makeStyles-contact-16 MuiTypography-body1 MuiTypography-alignLeft"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
        : 
        test@test.com
      </p>
      <p
        className="MuiTypography-root makeStyles-contact-16 MuiTypography-body1 MuiTypography-paragraph MuiTypography-alignLeft"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
          />
        </svg>
        : 
        <a
          href="https://test.com"
        >
           
          https://test.com
        </a>
      </p>
    </div>
  </div>
</div>
`;
